_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,205 @@
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- [H64: Using the title attribute of the frame and iframe elements](http://www.w3.org/TR/WCAG20-TECHS/H64.html)
+- [WCAG 2.0: Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)
+
+## Test cases
+
+### Passed
+
+#### Passed example 1
+
+Usage of `title` attribute to describe the `iframe` content, and there is only one iframe within document tree.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+```
+
+#### Passed example 2
+
+Multiple `iframe` elements in the document having different `title` descriptions as accessible name.
+
+```html
+<iframe title="List of Contributors to Repository 1" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors to Repository 2" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Passed example 3
+
+Multiple `iframe` elements in the document having different `aria-label` descriptions as accessible name.
+
+```html
+<iframe aria-label="List of Contributors to Repository 1" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors to Repository 2" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Passed example 4
+
+Multiple `iframe` elements in the document having different `aria-labelledby` descriptions as accessible name.
+
+```html
+<div id="desc-for-title">List of Contributors</div>
+<iframe aria-labelledby="desc-for-title" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<div id="desc-for-title1">List of Reviewers</div>
+<iframe aria-labelledby="desc-for-title1" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Passed example 5
+
+`iframes` having the same `title` within a given document tree, but one of them is not exposed to assistive technologies.
+
+```html
+<iframe aria-hidden='true' title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+
+#### Passed example 6
+
+`iframes` are allowed to have the same `title` across different document tree. In this example `iframe` with `id` `level2-frame1` has a parent document tree of `iframe` with `id` `level1-frame2`, and does not share the document tree of `iframe` with `id` `level1-frame1`.
+
+```html
+<iframe id="level1-frame1" title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe id="level1-frame2" title="List of Contributors 2" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+  <iframe id="level2-frame1" title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+  </iframe>
```

Comment:
I know what you're trying to do here, but this is not the way to express this in HTML, the content of an iframe element is use for its fallback, not to show the content of that frame. This content should actually be in the asset. We could put a comment inside the body of the iframe elm to show the content that's in the page it loads, but actually putting that content as a child element instead of inside the actual page doesn't work for me, and isn't going to work when we're someone runs their tool against this test case.